src/alpaka/AlpakaCore/alpakaWorkDivHelper.h

```diff
@@ -320,6 +320,384 @@ namespace cms {
       return isNextStrideElementValid;
     }
 
+    /*
+     * Class which simplifies "for" loops over elements index 
+     */
+    template <typename T, typename T_Acc>
+    class elements_with_stride {
```

Comment:
I missed the addition of `dimIndex` argument, nevermind.

Reply:
The idea is that `elements_with_stride` should loop over a single index using a scalar variable; usually this is the 0th index (assuming a one-dimensional kernel) but it can be chosen by `dimIndex`.
While `elements_with_stride_<N>d` should loop over an N-dimensional space using a `Vec3D` variable.

In fact, after having clarified that the platform and device are independent from the dimensionality, it makes sense to change `elements_with_stride_<N>d` to use a `Vec<N>D` instead of always a `Vec3D`.